include/distr_queue.h

```diff
@@ -86,6 +124,17 @@ class distr_queue {
 		}
 		tracker = std::make_shared<detail::distr_queue_tracker>();
 	}
+
+	detail::runtime::sync_guard slow_full_sync(detail::buffer_capture_map buffer_captures, detail::side_effect_map side_effects) {
+		check_not_drained(tracker);
+		return detail::runtime::get_instance().sync(std::move(buffer_captures), std::move(side_effects));
+	}
+
+	void drain(detail::buffer_capture_map buffer_captures, detail::side_effect_map side_effects) {
```

Comment:
I think this should be called something like `drain_internal`, to make things easier when reading the public `drain`s.

Reply:
Done.